src/dyad/common/dyad_flux_log.h

```diff
@@ -7,15 +7,57 @@
 #error "no config"
 #endif
 
+// clang-format off
 #include <sys/types.h>
-#include <unistd.h>
+#ifdef __cplusplus
+#include <cstdio>
+#else
+#include <stdio.h>
+#endif
+// clang-format on
+
 #include <flux/core.h>
+#include <unistd.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#if !defined(DYAD_LOGGING_ON) || (DYAD_LOGGING_ON == 0)
+#if defined(DYAD_LOGGING_ON) && (DYAD_LOGGING_ON == 1) && defined(DYAD_LOGGING_FLUX) \
+    && (DYAD_LOGGING_FLUX == 1)
+#define DYAD_LOG_INFO_REDIRECT(fpath) \
+    do {                              \
+    } while (0)
+#define DYAD_LOG_ERR_REDIRECT(fpath) \
+    do {                             \
+    } while (0)
+#define DYAD_LOG_INFO(dyad_ctx, ...) flux_log (dyad_ctx->h, LOG_INFO, __VA_ARGS__)
+#define DYAD_LOG_ERR(dyad_ctx, ...) flux_log_error (dyad_ctx->h, __VA_ARGS__)
+#define FLUX_LOG_INFO(flux_ctx, ...) flux_log (flux_ctx, LOG_INFO, __VA_ARGS__)
+#define FLUX_LOG_ERR(flux_ctx, ...) flux_log_error (flux_ctx, __VA_ARGS__)
+#elif defined(DYAD_LOGGING_ON) && (DYAD_LOGGING_ON == 1) && defined(DYAD_LOGGING_PRINTF) \
+    && (DYAD_LOGGING_PRINTF == 1)
+#define DYAD_LOG_INFO_REDIRECT(fpath) freopen (fpath, "a+", stdout)
+#define DYAD_LOG_ERR_REDIRECT(fpath) freopen (fpath, "a+", stderr)
+#define DYAD_LOG_INFO(dyad_ctx, ...) \
+    (fprintf (stdout, __VA_ARGS__),  \
+     fflush (stdout))
+#define DYAD_LOG_ERR(dyad_ctx, ...)  \
+    (fprintf (stderr, __VA_ARGS__),  \
+     fflush (stderr))
+#define FLUX_LOG_INFO(flux_ctx, ...) \
+    (fprintf (stdout, __VA_ARGS__),  \
+     fflush (stdout))
+#define FLUX_LOG_ERR(flux_ctx, ...)  \
+    (fprintf (stderr, __VA_ARGS__),  \
+     fflush (stderr))
```

Comment:
Multiple statements substituted by a macro should be grouped as one. Otherwise, it can be misused.
e.g.,
```
#MACRO_FUNC() statement1; statement2

if (CHK)
   MACRO_FUNC();
```
There are multiple options. I tried "(st1, st2)" option here but if this does not work try "do {st1; st2} while(0)"